content/post/danielepo/pills/gestire-le-versioni-con-gitversion-introduzione.it.md

```diff
@@ -0,0 +1,102 @@
+
+Incrementate la Patch quando fate piccole modifiche e correzioni di bug che non influenzato come il mondo esterno interagisce con il vostro programma o libreria
+
+Incrementare la Minor quando fate modifiche o aggiunte funzionalità senza però rompere la compatibilità con le versioni precedenti
+
+ed infine, incrementate la Major quando fate modifiche che rendono parti del nuovo codice non più compatibile con le versioni precedenti e quindi che richiederebbero modifiche da parte dei programmi che dipendono dalla vostra libreria 
+
+Seguendo questa convenzione un utilizzatore della vostra libreria riesce a identificare a colpo d'occhio se può aggiornare la versione senza sorprese o se invece gli conviene leggere attentamente i Change Log.
+
+Per esempio facciamo finta che ho una libreria che fa i calcoli di conversione tra varie monete e la versione di questa libreria è "uno cinque tre", mi segnalano che una stringa ha un errore, quando lo correggo creerò la versione "uno cinque quattro", incrementando di uno la patch.
+
+se invece sto creando una nuova funzionalità per esempio aggiungo una moneta alla lista di monete tra cui posso fare la conversione dovrò incrementare la Minor azzerando allo stesso tempo la Patch, creo quindi la versione uno sei zero.
+
+adesso per qualche strano motivo invertiamo l'ordine in cui passo i dati alla funzione di calcolo, invece di essere quantità, moneta di partenza, moneta di cui voliamo sapere il valore mettiamo le due monete come primi parametri e la quantità la passiamo per ultimo. questa modifica romperebbe il codice dei nostri utilizzatori e richiederebbe loro di modificare il loro codice. per questa ragione aggiorneremmo la Major azzerando minor e patch allo stesso tempo. quindi la nostra versione diventa due zero zero.
+
+spero sia chiaro quanto una convenzione di questo genere sia utile per gli sviluppatori che ora possono decidere se possono aggiornare liberamente le librerie o se è meglio che si prendano un po' di tempo per farlo per evitare sorprese. 
+
+Ora che conosciamo alcuni motivi per cui conviene gestire le versioni del software, in particolare delle librerie, possiamo vedere come automatizzare la creazione della versione grazie ad un programma chiamato gitversion. Come potete immaginare dal nome gitversion sfrutta git quindi se ancora non state usando git, beh, questo è un altro tra I tanti motivi per aggiornare il vostro sistema di controllo di versione software. 
+
+Per ora ci concentriamo sull'installazione e sull'utilizzo in locale e in un prossimo video vedremo come utilizzarlo nella nostra pipeline di build e rilascio.
+
+Nel sito di documentazione di gitversion questo viene definito semplicemente come uno strumento ti aiuta ad implementare il semantic versioning nei tuoi progetti. Sebbene questa reazione riassuma perfettamente l'utilità di gitversion, per meglio apprezzarlo è meglio vederlo in azione. Procediamo quindi ad installarlo
```

Comment:
- "strumento utile a implementare", non "strumento ti aiuta ad implementare"
- "Sebbene questa definizionre riassuma", non "Sebbene questa reazione riassuma"
- "per apprezzarlo al massimo è meglio vederlo in azione", non "per meglio apprezzarlo è meglio vederlo in azione"

Reply:
alcuni di questi errori non capisco come mi siano sfuggiti. Per esempio, il secondo punto... ho letto più volte **definizione** nelle varie registrazioni che ho fatto e non ho mai notato che c'era scritto **reazione**!